### PR TITLE
feat(cb2-8085): create get end point

### PR DESCRIPTION
**Pull request by naathanbrown**
## Description

Adding a lambda for the get end point. This also transforms the database response.

Related issue: [8085](https://dvsa.atlassian.net/browse/CB2-8085)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x ] Feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: src/util/formatTechRecord.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
+const buildArray = (techRecordWithoutArrays: object, arrayName: string, formattedTechRecord: any): object => {
```
Comment (naathanbrown) [inline src/util/formatTechRecord.ts:1]: The es-lint ignore will improve once we have correct types, so let's ignore for now as it will be fixed later :)

**Review by kcarr-bjss — APPROVED** [review]

File: src/handler/get.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> => {
+  logger.info('Get end point called');
+
+  const getErrors = getGetErrors(event);
+  if (getErrors) {
+    return addHttpHeaders(getErrors);
+  }
+
+  const systemNumber: string = decodeURIComponent(event.pathParameters?.systemNumber ?? '');
```
Comment (gjulien-bjss) [inline src/handler/get.ts:17]: Might want to validate the systemNumber and createdTimestamp are actually being sent (like the validate function we have on the search endpoint).

Comment (naathanbrown) [reply src/handler/get.ts:17]: NO

File: src/services/database.ts [hunk 3]
```diff
@@ -65,6 +67,25 @@ export const searchByAll = async (searchIdentifier: string): Promise<SearchResul
...
+      createdTimestamp,
+    }),
+  };
+
+  try {
+    const data = await ddbClient.send(new GetItemCommand(command));
+    logger.debug(JSON.stringify(data));
+    return unmarshall(data.Item || {});
+  } catch (e: any) {
+    logger.error('Error in search by sysnum and time: ', e);
```
Comment (gjulien-bjss) [inline src/services/database.ts:84]: stringify the error? (also I'm not sure the logger library accepts multiple arguments to log)

File: src/util/formatTechRecord.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      const splitKey = key.split('_');
+
+      if (parseInt(splitKey[2], 10) === arrayCount) {
+        splitKey.splice(1, 2);
+        const newKey = splitKey.join('_');
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, security/detect-object-injection, @typescript-eslint/no-unsafe-member-access
+        objectToAdd[newKey] = value;
+      } else {
+        array.push(objectToAdd);
+        arrayCount = parseInt(splitKey[2], 10);
```
Comment (gjulien-bjss) [inline src/util/formatTechRecord.ts:16]: `arrayIndex` ?

File: src/validators/get.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  const systemNumber: string = decodeURIComponent(event.pathParameters.systemNumber);
+  if (systemNumber.length < 3 || systemNumber.length > 21) {
+    return {
+      statusCode: 400,
+      body: 'The system number should be between 3 and 21 characters.',
+    };
+  }
+
+  const createdTimestamp = decodeURIComponent(event.pathParameters.createdTimestamp);
+  if (!/[0-9]+-[0-9]+-[0-9]+T[0-9]+:[0-9]+:[0-9]+\.[0-9]+Z/i.test(createdTimestamp)) {
```
Comment (gjulien-bjss) [inline src/validators/get.ts:28]:
```suggestion
  if (!/\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{3}Z/i.test(createdTimestamp)) {
```

**Review by gjulien-bjss — APPROVED** [review]

**Pull request merged**